app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,27 @@
+class RecipesController < ApplicationController
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = current_user.recipes.create(recipe_params)
+    if @recipe.save
+      flash[:notice] = "Recipe saved."
+      redirect_to @recipe
+    else
+      flash[:error] = "There was an error saving your recipe: #{@recipe.errors.full_messages.first}"
+      redirect_to new_recipe_path
```

Comment:
`render :new` only sets the view template to use: `new.html.haml`, it doesn't execute the `new` action.

You could call the `new` action here instead of `render :new`, though that would lose any data the user submitted as `@recipe` is re-initialized in `new`.

Instead we could try something along these lines, perhaps in the call today:

```ruby
# It's a view concern that we show the full number of open grain slots, so
# it's OK to have this logic in the template (and remove it from RecipesController#new):
@recipe.grains.remaining_capacity.times { @recipe.grains.build }



# More about Association Extensions:
# http://guides.rubyonrails.org/association_basics.html#association-extensions

# Add to existing has_many :grains ... relationship in Recipe model
has_many :grains, dependent: :destroy do
  
  # Return the number of open grain spots. Say we have 3 grains already
  # specified for the recipe, then `remaining_capacity` would be 2. (Assuming
  # max capacity is 5 grains).
  def remaining_capacity
    max_capacity = 5
    max_capacity - proxy_association.target.size
  end
end
```